feat: add database snapshot functionality to Helm chart (#9956)
determined-ai/determined: docs/reference/deploy/helm-config-reference.rst

      PersistentVolumes is disabled, users must manually create a PersistentVolume that will match
      the PersistentVolumeClaim.

   -  ``claimSuffix``: Optional configuration that defines the persistent volume claim name for the
      Determined database. If left undefined, a new PVC is created, named
      "determined-db-pvc-$releaseName". Otherwise, use the variable provided as the suffix to
      "determined-db-pvc-".

   -  ``snapshotSuffix``: Optional configuration of the volumeSnapshot name, a backup of the
      Determined's database persistentVolume. If defined, Helm will "snapshot" the DB upon its next
      upgrade, creating a volumeSnapshot called ``determined-db-snapsnot-$snapshotSuffix``. This
      CANNOT be the name of a persistent volume claim that already exists, like the
      ``determined-db-pvc-$releaseName``.

   -  ``restoreSnapshotSuffix``: Optional configuration of the volumeSnapshot name from which a new

>       persistentVolume will be created and swapped into Determined's database. This only applies

Reviewer: I did stumble a bit to understand this. Maybe it would help to start with a high-level description of what it's for and then dig into the implementation details. Something like:

> `restoreSnapshotSuffix`: Optional configuration of a volumeSnapshot to restore from during an upgrade. If this is set during an upgrade, a new persistentVolume will be created and swapped into Determined's database by creating a new persistent volume & claim named `determined-db-pvc-$restoreSnapshotSuffix` then restoring data from `determined-db-snapsnot-$restoreSnapshotSuffix`. MUST NOT be the name of a PVC that already exists, e.g. the `$releaseName`.

I don't have strong feelings about this, so up to you and what feels clearest to you.

      PersistentVolumes is disabled, users must manually create a PersistentVolume that will match
      the PersistentVolumeClaim.

   -  ``claimSuffix``: Optional configuration that defines the persistent volume claim name for the

>       Determined database. If left undefined, a new PVC is created, named

Reviewer 2: ```suggestion
Determined database. If not undefined, a new PVC is created, with the name
```

>       "determined-db-pvc-$releaseName". Otherwise, use the variable provided as the suffix to

Reviewer 2: ```suggestion
"determined-db-pvc-$releaseName". If specified, the provided value will be used as the suffix to "determined-db-pvc-".
```

>       "determined-db-pvc-".

Reviewer 2: ```suggestion
If specified, the provided value will be used as the suffix to "determined-db-pvc-".
```

>    -  ``snapshotSuffix``: Optional configuration of the volumeSnapshot name, a backup of the

Reviewer 2: ```suggestion
- ``snapshotSuffix``: Optional configuration for naming the volume snapshot, which serves as a backup of the
```

>       Determined's database persistentVolume. If defined, Helm will "snapshot" the DB upon its next

Reviewer 2: ```suggestion
Determined database's persistentVolume. If defined, Helm will create a snapshot of the database during the next
```

      upgrade, creating a volumeSnapshot called ``determined-db-snapsnot-$snapshotSuffix``. This

>       MUST NOT be the name of a persistent volume claim that already exists, such as

Reviewer 2: ```suggestion
**must** **not** match the name of an existing persistent volume claim, such as
```

>       ``determined-db-pvc-$releaseName`` because the ``snapshotSuffix`` will create a new persistent

Reviewer 2: ```suggestion
``determined-db-pvc-$releaseName``, because the ``snapshotSuffix`` will be used to create a new persistent
```

>       volume claim when restored.

Reviewer 2: ```suggestion
 volume claim when restoring.
```

   -  ``restoreSnapshotSuffix``: Optional configuration of the volumeSnapshot to restore from during
      an upgrade. If this is set during an upgrade, a new persistentVolume will be created and
      swapped into Determined's database by creating a new persistent volume claim named
      ``determined-db-pvc-$restoreSnapshotSuffix``, restoring data from

>       ``determined-db-snapsnot-$restoreSnapshotSuffix``. MUST NOT be the name of a PVC that already

Reviewer 2: ```suggestion
``determined-db-snapsnot-$restoreSnapshotSuffix``. must not match the name of an existing persistent volume claim (PVC), such as ``$releaseName``.
```

>       exists, e.g. the ``$releaseName``.

Reviewer 2: ```suggestion
```
